project/Build.scala

```diff
@@ -932,12 +932,14 @@ object Build extends sbt.Build {
           })),
 
           sources in Test ++= {
-            if (!scalaVersion.value.startsWith("2.10") &&
-                scalacOptions.value.contains("-Xexperimental")) {
-              (((sourceDirectory in Test).value / "require-sam") ** "*.scala").get
-            } else {
-              Nil
-            }
+            def listAllScalaFilesIf(testDir: String, condition: Boolean): Seq[File] =
+              if (condition) (((sourceDirectory in Test).value / testDir) ** "*.scala").get
+              else Nil
+            val javaVersion = System.getProperty("java.version")
+            listAllScalaFilesIf("require-jdk7", javaVersion.matches("1\\.[78].*")) ++
+            listAllScalaFilesIf("require-jdk8", javaVersion.startsWith("1.8")) ++
+            listAllScalaFilesIf("require-sam", !scalaVersion.value.startsWith("2.10") &&
+              scalacOptions.value.contains("-Xexperimental"))
```

Comment:
Indentation issue (should be 4).
Also, I would recommend breaking the line already before `!scalaVersion`, because as it stands, I read `&&` as being on the same level as `++`, which is weird.